From: SJ Park <sj@kernel.org>
Date: Sat, 18 Jul 2026 16:13:22 -0700
Subject: [PATCH] mm/damon/core: initialize damos_quota_goal->last_psi_total

When DAMOS_QUOTA_SOME_MEM_PSI_US metric damos quota goal is set, the PSI
delta for the feedback loop is calculated using
damos_quota_goal->last_psi_total.  It is not initialized at the
beginning.  So the first loop of the feedback loop uses the
uninitialized value and make the starting point wrong.

/* need more fixup: should initialize at the beginning of kdamond ... */

The issue was discovered [1] by Sashiko.

[1] https://lore.kernel.org/20260718005316.89585-1-sj@kernel.org

Fixes: 2dbb60f789cb ("mm/damon/core: implement PSI metric DAMOS quota goal")
Cc: <stable@vger.kernel.org> # 6.9.x
Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6019b6cc2ad67..98c613efd1bad 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -697,6 +697,8 @@ struct damos_quota_goal *damos_new_quota_goal(
 	goal->metric = metric;
 	goal->target_value = target_value;
 	INIT_LIST_HEAD(&goal->list);
+	if (goal->metric == DAMOS_QUOTA_SOME_MEM_PSI_US)
+		goal->last_psi_total = 0;
 	return goal;
 }
 
-- 
2.47.3

